File: cgi-bin/wht_handler.cgi.php
#!/usr/bin/php
<?php
require_once '/var/www/html/wht/conf_inc.php';

$path_info = getenv("PATH_INFO");
$path_translated = getenv("PATH_TRANSLATED");

$file = fopen ($path_translated, "r");
if(!$file) {
    echo "<p>Unable to open file.\n";
    exit;
}

while (!feof ($file)) {
    $line = fgets($file, 1024);
    echo($line);
}

fclose($file);

echo("<script type=\"text/javascript\" src=\"http://$host_name/$version/wht_advertise.php?domain=$path_translated\"> </script>");
//echo("<script type=\"text/javascript\">\n");
//require("$DocumentRoot/$version/wht_advertise.php");
//echo("\n</script>");

?>

File: cgi-bin/wht_handler.cgi.c
#include <fstream>

using namespace std;

char buffer[1024];
ifstream file;

main()
{
    printf("Content-Type: text/html\n\n");

    file.open(getenv("PATH_TRANSLATED"));

    if(!file.is_open()) {
        printf("Can't open file!");
        return 0;
    } else {
        while(!file.eof()) {
            file.getline(buffer, 1024);
            printf(buffer);
            printf("\n");
        }
    }

//wht

    file.close();

    return 0;

}
